File: pythonbrew/commands/help.py
from pythonbrew.basecommand import Command, command_dict
from pythonbrew.baseparser import parser
from pythonbrew.log import logger

class HelpCommand(Command):
    name = "help"
    usage = "%prog [COMMAND]"
    summary = "Show available commands"
    
    def run_command(self, options, args):
        if args:
            command = args[0]
            if command not in command_dict:
                parser.error("Unknown command: `%s`" % command)
                return
            command = command_dict[command]
            command.parser.print_help()
            return
        parser.print_help()
        logger.log("\nCommands available:")
        commands = [command_dict[key] for key in sorted(command_dict.keys())]
        for command in commands:
            logger.log("  %s: %s" % (command.name, command.summary))
        logger.log("\nFurther Instructions:")
        logger.log("  https://github.com/utahta/pythonbrew")

HelpCommand()

File: subcmd/help.go
package subcmd

import (
	"fmt"
)

// Help command
type Help struct{}

// NewHelp returns Help
func NewHelp() *Help {
	return &Help{}
}

// Name returns command name
func (c *Help) Name() string {
	return "help"
}

// Summary returns command summary
func (c *Help) Summary() string {
	return "Show commands"
}

// Usage shows command usage
func (c *Help) Usage() {
	fmt.Printf(`Usage: pythonbrew COMMAND [OPTIONS]
  -h --help
    Show commands
  -v --version
    Show version

Commands:
`)
	for _, c := range Repository().Commands() {
		if c.Name() == "init" || c.Name() == "help" {
			continue
		}
		fmt.Printf("  %-9s  %s\n", c.Name(), c.Summary())
	}
	fmt.Printf(`
See more details:
  pythonbrew COMMAND -h
`)
}

// Run runs help command
func (c *Help) Run(args []string) error {
	c.Usage()
	return nil
}
